include/boost/json/detail/value_from.hpp

```diff
@@ -148,7 +152,50 @@ value_from_helper(
         "No suitable tag_invoke overload found for the type");
 }
 
+#ifndef BOOST_NO_CXX17_HDR_OPTIONAL
+template<class>
+struct is_optional
+    : std::false_type
+{ };
+
+template<class T>
+struct is_optional< std::optional<T> >
+    : std::true_type
+{ };
+#endif
+
 } // detail
+
+#ifndef BOOST_NO_CXX17_HDR_OPTIONAL
+template<class T>
+typename std::enable_if<
+    detail::is_optional< detail::remove_cvref<T> >::value
+    >::type
+tag_invoke(
+    value_from_tag,
+    value& jv,
+    T&& from)
+{
+    using InnerRef = decltype( *static_cast<T>(from) );
+    if( from )
+        value_from(static_cast<InnerRef>(*from), jv);
+    else
+        jv = nullptr;
+}
+
+inline
+void
+tag_invoke(
+    value_from_tag,
+    value& jv,
+    std::nullopt_t)
+{
```

Comment:
Shouldn't `nullopt_t` be marked as `is_null_like` instead?

Reply:
The generic implementation does not work, because std::nullopt_t is not default constructible.